webserver/views/api/v1/datasets.py

```diff
@@ -136,18 +137,47 @@ def add_class(dataset_id):
         {
             "name": "Not Mood",
             "description": "Dataset for mood misclassification.",
+            "recordings": ["770cc467-8dde-4d22-bc4c-a42f91e"]
         }
 
     :reqheader Content-Type: *application/json*
     :<json string name: *Required.* Name of the class. Must be unique within a dataset.
     :<json string description: *Optional.* Description of the class.
-    :<json array public: *Optional.* Array of recording MBIDs (``string``) to add into that class. For example:
+    :<json array recordings: *Optional.* Array of recording MBIDs (``string``) to add into that class. For example:
         ``["770cc467-8dde-4d22-bc4c-a42f91e"]``.
 
 
     :resheader Content-Type: *application/json*
     """
-    raise NotImplementedError
+    ds = get_check_dataset(dataset_id)
+    if ds["author"] != current_user.id:
+        raise api_exceptions.APIUnauthorized("You can't create this class.")
+    class_dict = request.get_json()
+    if not class_dict:
+        raise api_exceptions.APIBadRequest("Data must be submitted in JSON format.")
+    if "name" not in class_dict:
+        raise api_exceptions.APIBadRequest("name key missing in JSON request.")
+    if class_dict["name"] in (classes["name"] for classes in ds["classes"]):
+        raise api_exceptions.APIBadRequest("Class already exists.")
+    if "recordings" in class_dict:
+        for mbid in class_dict["recordings"]:
+            try:
+                UUID(mbid, version=4)
+            except ValueError:
+                raise api_exceptions.APIBadRequest("MBID %s not a valid UUID" % (mbid,))
+        unique_mbids = list(set(class_dict["recordings"]))
+        class_dict["recordings"] = unique_mbids
+    try:
+        db.dataset.add_class(class_dict, dataset_id)
```

Comment:
You are already checking if this dataset exists, using `get_check_dataset`. I think at this point you can assume that it exists, and that this `add_class` method won't fail to find the dataset.